Lib/fontTools/varLib/errors.py

```diff
@@ -32,40 +16,146 @@ def __init__(self, merger, args):
-        ]
-        context = "".join(reversed(stack))
-        details = ""
-        reason = cause["reason"].value
-        if reason == VarLibMergeFailure.FoundANone:
-            offender = [x is None for x in cause["got"]].index(True)
-            details = (
-                f"\n\nThe problem is likely to be in {fontnames[offender]}:\n"
-                f"{stack[0]}=={cause['got']}\n"
-            )
-        elif "expected" in cause and "got" in cause:
-            offender = [x == cause["expected"] for x in cause["got"]].index(False)
-            got = cause["got"][offender]
-            details = (
-                f"\n\nThe problem is likely to be in {fontnames[offender]}:\n"
-                f"Expected to see {stack[0]}=={cause['expected']}, instead saw {got}\n"
-            )
+    @property
+    def cause(self):
+        return self.args[0]
+
+    @property
+    def stack(self):
+        return self.args[1:]
+
+    @property
+    def reason(self):
+        return self.__doc__
 
+    def _master_name(self, ix):
+        ttf = self.merger.ttfs[ix]
         if (
-            reason == VarLibMergeFailure.UnsupportedFormat
-            or reason == VarLibMergeFailure.InconsistentFormat
+            "name" in ttf
+            and ttf["name"].getDebugName(1)
+            and ttf["name"].getDebugName(2)
         ):
-            reason = reason % cause["subtable"]
+            return ttf["name"].getDebugName(1) + " " + ttf["name"].getDebugName(2)
+        elif hasattr(ttf.reader, "file") and hasattr(ttf.reader.file, "name"):
+            return ttf.reader.file.name
+        else:
+            return "master number %i" % ix
+
+    @property
+    def offender(self):
+        if "expected" in self.cause and "got" in self.cause:
+            index = [x == self.cause["expected"] for x in self.cause["got"]].index(
+                False
+            )
+            return index, self._master_name(index)
+        return None, None
+
+    @property
+    def details(self):
+        if "expected" in self.cause and "got" in self.cause:
+            offender_index, offender = self.offender
+            got = self.cause["got"][offender_index]
+            return f"Expected to see {self.stack[0]}=={self.cause['expected']}, instead saw {got}\n"
+        return ""
+
+    def __str__(self):
+        offender_index, offender = self.offender
+        location = ""
+        if offender:
+            location = f"\n\nThe problem is likely to be in {offender}:\n"
+        context = "".join(reversed(self.stack))
         basic = textwrap.fill(
-            f"Couldn't merge the fonts, because {reason}. "
+            f"Couldn't merge the fonts, because {self.reason}. "
             f"This happened while performing the following operation: {context}",
             width=78,
         )
-        return "\n\n" + basic + details
+        return "\n\n" + basic + location + self.details
+
+
+class VarLibMergeFailureShouldBeConstant(VarLibMergeError):
```

Comment:
do we actually need such lengthy exception class names? They are defined in varLib.errors so they are already scoped in a way. Just the name of the failure is enough to identify them as "VarLibMergeFailures".
If you like just keep a prefix like "MergeError*" (Error being shorter than Failure)

Reply:
the base class "VarLibMergeError" is different in that it pre-existed the new subclasses and we can't easily change its name now